pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -632,7 +640,7 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*re
 			continue
 		}
 		klog.V(4).Infof("Volume: %s (pod UID %s) is marked as mounted and added into the actual state", volume.volumeName, volume.podName)
-		if volume.attachablePlugin != nil {
+		if volume.attachablePlugin != nil || volume.blockVolumeMapper != nil {
```

Comment:
local volume does not implement AttachableVolumePlugin but we need device path returned from `blockVolumeMapper.GetGlobalMapPath`

Reply:
I would appreciate if the information above was captured as comment in the code, so we don't forget why it's the way it is.

Reply:
done
it applies for all volume plugins which didn't implement `AttachableVolumePlugin` but implemented `BlockVolumeMapper`

Reply:
It should be deviceMountable plugin (or check devideMounter). I will fix it in a follow PR which is going to support block fs local volume reconstruction. less change in this pr.

Reply:
Can we also test a non-attachable plugin that implements block, like iscsi, to make sure this change is ok for them?

Reply:
This succeeded: `[sig-storage] In-tree Volumes [Driver: iscsi][Feature:Volumes] [Testpattern: Pre-provisioned PV (block volmode)] disruptive[Disruptive] Should test that pv used in a pod that is force deleted while the kubelet is down cleans up when the kubelet returns.`

Reply:
Thanks!

Reply:
hi, I've checked all plugins, `iscsi` [implements AttachableVolumePlugin](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/iscsi/attacher.go#L45). Right now only local volume does not implement AttachableVolumePlugin but implements BlockVolumePlugin. So this code change only affects local volume.